cmd/fluxctl/build.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/spf13/cobra"
+	disco "k8s.io/client-go/discovery"
+	"k8s.io/client-go/tools/clientcmd"
+
+	kube "github.com/fluxcd/flux/pkg/cluster/kubernetes"
+	"github.com/fluxcd/flux/pkg/manifests"
+)
+
+type buildOpts struct {
+	base       string
+	paths      []string
+	generation bool   // use manifest generation
+	defaultNS  string // override the default namespace
+}
+
+func newBuild() *buildOpts {
+	return &buildOpts{}
+}
+
+func (opts *buildOpts) Command() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "build",
+		Short: "Generate manifests from a local directory as fluxd would",
+		RunE:  opts.RunE,
+	}
+
+	cmd.Flags().StringVarP(&opts.base, "base", "C", ".", "run as if invoked in the directory given; the working directory is treated as the root, for the purpose of interpreting config")
+	cmd.Flags().StringSliceVarP(&opts.paths, "path", "p", nil, "use the path(s) given as target paths to be examined; if not supplied, the root directory is assumed as the sole target path")
+	cmd.Flags().BoolVarP(&opts.generation, "manifest-generation", "g", false, "act the same as fluxd with --manifest-generation; i.e., look for .flux.yaml files in the target path(s) and use the instructions there")
+	cmd.Flags().StringVar(&opts.defaultNS, "default-namespace", "", "give the default namespace explicitly, rather than taking it from KUBE_CONFIG")
```

Comment:
we should make this `--k8s-default-namespace` to make it consistent with `fluxd`